### PR TITLE
fix: respect inner attributes for Structs and Enums

### PR DESCRIPTION
**Pull request by rainy-me**
fix: #10980 (the allow/deny issue is not fully resolved though.)

### CODE REVIEW

File: crates/hir_ty/src/diagnostics/decl_check.rs [hunk 1]
```diff
@@ -181,8 +181,12 @@ impl<'a> DeclValidator<'a> {
                 AttrDefId::ExternBlockId(id) => Some(id.lookup(self.db.upcast()).container.into()),
                 // These warnings should not explore macro definitions at all
                 AttrDefId::MacroDefId(_) => None,
-                // Will never occur under an enum/struct/union/type alias
```
Comment (rainy-me) [inline crates/hir_ty/src/diagnostics/decl_check.rs:184]: I think this pr should be the fix but have no idea why origin comment says this.

Comment (Veykril) [reply crates/hir_ty/src/diagnostics/decl_check.rs:184]: Ye that comment seems to be wrong

**Comment by Veykril** [comment]
Thanks!
bors r+

**Pull request merged**